### PR TITLE
Upgrade otel collector core, fix changes after no nil values in slices

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by tigrannajaryan — APPROVED** [review]

**Comment by axw** [comment]
Came here from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1735#issuecomment-736852408. FWIW, Elastic changes look good. Thanks!

File: processor/groupbytraceprocessor/processor_test.go [hunk 1]
```diff
@@ -179,15 +179,10 @@ func TestProcessBatchDoesntFail(t *testing.T) {
 	span := ils.Spans().At(0)
 	span.SetTraceID(traceID)
 	span.SetSpanID(pdata.NewSpanID([8]byte{1, 2, 3, 4}))
-	batch.ResourceSpans().Append(pdata.NewResourceSpans())
 
 	p, err := newGroupByTraceProcessor(logger, st, next, config)
 	require.NoError(t, err)
 
-	// sanity check
-	err = p.processResourceSpans(batch.ResourceSpans().At(1))
```
Comment (jpkrohling) [inline processor/groupbytraceprocessor/processor_test.go:188]: Why was this removed? This was a sanity check...

**Review by jpkrohling — COMMENTED** [review]
Looks good for most of the modules I'm a code owner, but I have a question about a removed assertion.

**Comment by djaglowski** [comment]
`stanzareceiver` looks good.

File: processor/groupbytraceprocessor/processor_test.go (hunk 1 above)
Comment (jpkrohling) [reply processor/groupbytraceprocessor/processor_test.go:188]: ping @bogdandrutu

Comment (bogdandrutu) [reply processor/groupbytraceprocessor/processor_test.go:188]: Sorry, definitely missed this, looks like test does not exist anymore
